java/impl/aws/src/main/java/co/worklytics/psoxy/S3Handler.java

```diff
@@ -71,21 +72,18 @@ StorageEventResponse process(String importBucket, String sourceKey, StorageHandl
             return null;
         }
 
+        S3Object s3Object = s3Client.getObject(new GetObjectRequest(importBucket, sourceKey));
         try (InputStream objectData = s3Object.getObjectContent();
              BOMInputStream is = new BOMInputStream(objectData);
-             InputStreamReader reader = new InputStreamReader(is, StandardCharsets.UTF_8)) {
+             InputStreamReader reader = new InputStreamReader(is, StandardCharsets.UTF_8);
+             PipedOutputStream outputStream = new PipedOutputStream();
+             OutputStreamWriter writer = new OutputStreamWriter(outputStream);
```

Comment:
> or top efficiency, consider wrapping an OutputStreamWriter within a BufferedWriter so as to avoid frequent converter invocations. For example:
> Writer out  = new BufferedWriter(new OutputStreamWriter(System.out));
https://docs.oracle.com/javase/8/docs/api/java/io/OutputStreamWriter.html